contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,893 @@
+      tokenLocking.claim(token, true);
+
+      emit PollEventSet(_pollId, STAKE_END);
+    }
+
+    emit PollStaked(_pollId, msg.sender, _vote, amount);
+  }
+
+  /// @notice Submit a vote secret for a poll
+  /// @param _pollId The id of the poll
+  /// @param _voteSecret The hashed vote secret
+  /// @param _key Reputation tree key for the staker/domain
+  /// @param _value Reputation tree value for the staker/domain
+  /// @param _branchMask The branchmask of the proof
+  /// @param _siblings The siblings of the proof
+  function submitVote(
+    uint256 _pollId,
+    bytes32 _voteSecret,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Submit, "voting-rep-poll-not-open");
+    require(_voteSecret != bytes32(0), "voting-rep-invalid-secret");
+
+    uint256 userRep = getReputationFromProof(_pollId, msg.sender, _key, _value, _branchMask, _siblings);
+
+    // Count reputation if first submission
+    if (voteSecrets[_pollId][msg.sender] == bytes32(0)) {
+      poll.repSubmitted = add(poll.repSubmitted, userRep);
+    }
+
+    voteSecrets[_pollId][msg.sender] = _voteSecret;
+
+    emit PollVoteSubmitted(_pollId, msg.sender);
+
+    if (poll.repSubmitted >= wmul(poll.skillRep, maxVoteFraction)) {
+      poll.events[SUBMIT_END] = uint64(now);
+      poll.events[REVEAL_END] = uint64(now + revealPeriod);
+
+      emit PollEventSet(_pollId, SUBMIT_END);
+    }
+  }
+
+  /// @notice Reveal a vote secret for a poll
+  /// @param _pollId The id of the poll
+  /// @param _salt The salt used to hash the vote
+  /// @param _vote The side being supported (0 = NAY, 1 = YAY)
+  /// @param _key Reputation tree key for the staker/domain
+  /// @param _value Reputation tree value for the staker/domain
+  /// @param _branchMask The branchmask of the proof
+  /// @param _siblings The siblings of the proof
+  function revealVote(
+    uint256 _pollId,
+    bytes32 _salt,
+    uint256 _vote,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Reveal, "voting-rep-poll-not-reveal");
+
+    uint256 userRep = getReputationFromProof(_pollId, msg.sender, _key, _value, _branchMask, _siblings);
+
+    bytes32 voteSecret = voteSecrets[_pollId][msg.sender];
+    require(voteSecret == getVoteSecret(_salt, _vote), "voting-rep-secret-no-match");
+    delete voteSecrets[_pollId][msg.sender];
+
+    poll.votes[_vote] = add(poll.votes[_vote], userRep);
+    poll.repRevealed = add(poll.repRevealed, userRep);
+
+
+    uint256 fractionUserReputation = wdiv(userRep, poll.skillRep);
+    uint256 totalStake = add(poll.stakes[YAY], poll.stakes[NAY]);
+    uint256 voterReward = wmul(wmul(fractionUserReputation, totalStake), voterRewardFraction);
```

Comment:
In the case of an escalation, you need to take in to account any tokens that have been paid out in a previous poll (easiest to see if you consider `voterRewardFraction` being 0.5), otherwise you risk not being able to pay the stakers back or, in an extreme case, even all the voters. Fortunately, we're already tracking `paidVoterComp` so this should be easy.

Reply:
Mm you're right. I thought I'd handled this but unless we make voter comp relative, or require stakers to "top up" voter comp during escalations, we could have problems. I'd be in favor of requiring every escalation round to top up the voter comp, which seems to cleanly avoid the issue.